src/App.tsx

```diff
@@ -14,11 +14,106 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+type State = {
+  selectedGoods: string[];
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: [],
+  };
+
+  addInitialText = () => {
+    const { selectedGoods } = this.state;
+
+    switch (selectedGoods.length) {
+      case 0:
+        return 'No goods selected';
+      case 1:
+        return `${selectedGoods.join(', ')} is selected`;
+      default: return `${selectedGoods.join(', ')} are selected`;
```

Comment:
should be in format `a, b and c are selected` you're having `a, b, c are selected`